numcodecs/blosc.pyx

```diff
@@ -366,30 +374,38 @@ def decompress(source, dest=None):
     """
     cdef:
         int ret
-        char *source_ptr
-        char *dest_ptr
-        Buffer source_buffer
-        Buffer dest_buffer = None
+        memoryview source_mv
+        const Py_buffer* source_pb
+        const char* source_ptr
+        memoryview dest_mv
+        Py_buffer* dest_pb
+        char* dest_ptr
         size_t nbytes, cbytes, blocksize
 
-    # setup source buffer
-    source_buffer = Buffer(source, PyBUF_ANY_CONTIGUOUS)
-    source_ptr = source_buffer.ptr
+    # obtain source memoryview
+    source_mv = memoryview(source)
+    source_pb = PyMemoryView_GET_BUFFER(source_mv)
+    if not PyBuffer_IsContiguous(source_pb, b'A'):
+        raise BufferError("`source` must contain contiguous memory")
```

Comment:
this same sequence is repeated many times. do you think it would make sense to put it in a function?

Reply:
It's a fair question. Think the best we could get is dropping the contiguous check throughout (saving 2 lines)

Am ok either way

Reply:
the highlighted text is repeated verbatim 9 times in this PR, so I think replacing it with invocations of a function like
```
def get_memory_view(source) -> tuple[memory view, whatever source pb is]:
      source_mv = memoryview(source)
    source_pb = PyMemoryView_GET_BUFFER(source_mv)
    if not PyBuffer_IsContiguous(source_pb, b'A'):
        raise BufferError("`source` must contain contiguous memory")
   return source_mv, source_pb
```

would result in a lot of deduplication. but I don't know cython, so maybe there's some aspect of that that I'm missing?

Reply:
seems like a bit of a chore then. up to you if you deem it worthwhile

Reply:
Am ok punting on this

Reply:
Went ahead and added it. Maybe it is useful to have this in a standard function